codes/Mediatype.ttl

```diff
@@ -0,0 +1,14 @@
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix ids: <https://w3id.org/idsa/core/> .
+@prefix idsc: <https://w3id.org/idsa/code/> .
+
+<https://www.iana.org/assignments/media-types/application/json>
```

Comment:
Why did we introduce this new media type? For the other media types in the IANA namespace I thought we do not define themselves but simply write down the IANA URIs. So why do we give our own definition of this one? Because it does not (yet?) exist at IANA? No, actually it does exist: https://www.iana.org/assignments/media-types/application/json
In any case I'm not sure we should use a IANA URI. That's bad practice (namespace hijacking). I'd rather use an IDS infomodel URI. If we want to add our own information to that subject, we can do so, and say that our subject is `owl:sameAs` the IANA one.

Reply:
>Why did we introduce this new media type?

This is a necessary adjustment for our tools (Java Code Gen.) to work properly. Our code-gen tool behaves differently depending on whether a class is intended to contain instance data or not. Therefore, the code-gen tools needs instance data for this given class (media type) to transform it into the corresponding java class. 

**Idea:** We could put this information to an "external" file in another directory. A file with additional information, which are not really meant to be part of the ontology but required for our tooling. Statements like these could be defined in there.

---

> If we want to add our own information to that subject, we can do so, and say that our subject is owl:sameAs the IANA one.

Is your suggestion to add this information to the serialized instanes which are exchanged between connectors in the IDS as well ? I am not 100% sure, what the implications of this idea are. As far as I know, the code-gen / serializer should not really care what URI is used. 

It would still be some kind of a breaking change. Until now, we said the one should use the IANA URIs and label it as an `ids:IANAMediaType`. This has "spared" us from having to maintain our own vocabularies which are just copies of existing vocabularies. IANA type lists may be expanded over time, which we would have to track as well. 
For languages , we already did what you suggest. We also added the whole list of languages to our code list:
https://github.com/International-Data-Spaces-Association/InformationModel/blob/9afd5b4c0bfa5db46793c33aefff81a6a8e304ab/codes/Language.ttl#L18-L21
due to the fact that the list of languages are more or less fixed. This "issue" with media types mainly comes from our design decision that the media type should refer to a subclass of `ids:MediaType` (namely: `ids:IANAMediaType` / `ids:CustomMediaType`) and not to some URI. The idea behind that was that it would be easyier for a component to know what to do by checking the `@type` of a mediatype URI. `DO x IF IANAMediaType, DO y IF ids:CustomMediaType`

Changing the IANA type URIs to infomodel URIs would mean:

- Use infomodel URI  `https://w3id.org/idsa/code/....` following the media type (application/json). We have to take care in some way that everyone uses the same URI. E.g., define that `type/subtype` of the IANA should be used kept with the Infomdeol URI:  `<https://www.iana.org/assignments/media-types/application/json>` ->  `https:/w3id.org/idsa/code/application/json` 
- State that one should use `owl:sameAs <https://www.iana.org/assignments/media-types/application/json>` in the serialized JSON-LD (if I understand your suggestion correctly). Our tools have to support this as well. 

Maybe @sebbader has an opinion on this.